lib/rules/no-link-to-positional-params.js

```diff
@@ -6,62 +6,58 @@ module.exports = class NoLinkToPositionalParams extends Rule {
 
       this.log({ message, node });
     }
   }
 
-  getMessage(node) {
-    function getError(message) {
-      return `Invoking the \`<LinkTo>\` component with positional arguments is deprecated. ${message}`;
+  getMessage(node, { hasRequiredTextParam }) {
+    let requiredParamsCount = hasRequiredTextParam ? 2 : 1; // Route name is always required, text only for inline `{{link-to}}` components.
+    let hasQueryParams = includesQueryParams(node.params);
+    let modelsCount = countModelParams(node.params, requiredParamsCount, hasQueryParams);
+    let namedArguments = ['`@route`'];
+    let additionalInfo = '';
+
+    if (modelsCount === 1) {
+      namedArguments.push('`@model`');
+    } else if (modelsCount > 1) {
+      namedArguments.push('`@models`');
     }
 
-    if (node.params.length === 1 && node.params[0].type === 'StringLiteral') {
-      return getError('Instead, please use the equivalent named arguments (`@route`).');
+    if (hasQueryParams) {
+      namedArguments.push('`@query` using the `hash` helper');
     }
 
-    if (node.params.length === 2) {
-      if (node.params.every((param) => param.type === 'StringLiteral')) {
-        return getError(`Instead, please use the equivalent named arguments (\`@route\`) and pass a
-  block for the link's content.`);
-      }
-
-      if (node.params[0].type === 'StringLiteral' && node.params[1].type === 'PathExpression') {
-        return getError('Instead, please use the equivalent named arguments (`@route`, `@model`).');
-      }
-
-      if (
-        node.params[0].type === 'StringLiteral' &&
-        node.params[1].type === 'SubExpression' &&
-        node.params[1].path.original === 'query-params'
-      ) {
-        return getError(`Instead, please use the equivalent named arguments (\`@route\`, \`@query\`) and the
-\`hash\` helper.`);
-      }
+    if (hasRequiredTextParam) {
+      additionalInfo = ' The content should be passed along as a block.';
     }
 
-    if (node.params.length === 3) {
-      if (
-        node.params[0].type === 'StringLiteral' &&
-        node.params[1].type === 'PathExpression' &&
-        node.params[2].type === 'PathExpression'
-      ) {
-        return getError(
-          'Instead, please use the equivalent named arguments (`@route`, `@models`).'
-        );
-      }
-    }
+    return getErrorMessage(namedArguments.join(', '), additionalInfo);
   }
 };
+
+function includesQueryParams(params) {
+  return params.some((param) => {
+    return param.path && param.path.original === 'query-params';
+  });
+}
+
+function countModelParams(params, requiredCount, hasQueryParams) {
+  return Math.max(params.length - (hasQueryParams ? requiredCount + 1 : requiredCount), 0);
+}
+
+function getErrorMessage(namedArguments, additionalInfo) {
+  return `Invoking the \`<LinkTo>\` component with positional arguments is deprecated. Instead, please use the equivalent named arguments (${namedArguments}).${additionalInfo}`;
```

Comment:
Was wondering if it would be better if we use `{{link-to}}` here as that would better match the user's code?

Reply:
Ya, we should probably detect which variant they're using and output that. What do you think?

Reply:
We could probably just use `{{link-to}}`, but I guess we should then also omit the `@` char for the named arguments?

Reply:
Ya it would be nice to try to replicate what was matched on in the error message, so it's as accurate as possible. Can we do that?